src/core/announcements/validation.ts

```diff
@@ -0,0 +1,206 @@
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.GraphChange) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isGraphChangeType(obj["changeType"])) return false;
+  if (!isDSNPUserId(obj["objectId"])) return false;
+
+  return true;
+};
+
+/**
+ * isBroadcastAnnouncement() is a type check for BroadcastAnnouncement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a BroadcastAnnouncement, otherwise false
+ */
+const isBroadcastAnnouncement = (obj: unknown): obj is BroadcastAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Broadcast) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isString(obj["url"])) return false;
+  if (!isString(obj["contentHash"])) return false;
+
+  return true;
+};
+
+/**
+ * isReplyAnnouncement() is a type check for ReplyAnnouncement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a ReplyAnnouncement, otherwise false
+ */
+const isReplyAnnouncement = (obj: unknown): obj is ReplyAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Reply) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isString(obj["url"])) return false;
+  if (!isString(obj["contentHash"])) return false;
+  if (!isDSNPAnnouncementId(obj["inReplyTo"])) return false;
+
+  return true;
+};
+
+/**
+ * isReactionAnnouncement() is a type check for ReactionAnnouncement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a ReactionAnnouncement, otherwise false
+ */
+const isReactionAnnouncement = (obj: unknown): obj is ReactionAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Reaction) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isValidEmoji(obj["emoji"])) return false;
+  if (!isDSNPAnnouncementId(obj["inReplyTo"])) return false;
+
+  return true;
+};
+
+/**
+ * isProfileAnnouncement() is a type check for ProfileAnnouncement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a ProfileAnnouncement, otherwise false
+ */
+const isProfileAnnouncement = (obj: unknown): obj is ProfileAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Profile) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isString(obj["url"])) return false;
+  if (!isString(obj["contentHash"])) return false;
+
+  return true;
+};
+
+/**
+ * isAnnouncement() is a type check for Announcement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a Announcement, otherwise false
+ */
+const isAnnouncement = (obj: unknown): obj is Announcement => {
+  if (!isRecord(obj)) return false;
+  if (!isAnnouncementType(obj["announcementType"])) return false;
+
+  const validators = {
```

Comment:
We need to add this type so that TypeScript will throw an error if we add a new `AnnouncementType`
```suggestion
  const validators: Record<AnnouncementType, (obj: unknown) => boolean> = {
```